extensions/amp-geo/0.1/amp-geo.js

```diff
@@ -188,20 +188,18 @@ export class AmpGeo extends AMP.BaseElement {
     const trimmedGeoMatch = GEO_HOTPATCH_STR_REGEX.exec(COUNTRY);
 
     // default country is 'unknown' which is also the zero length case
-    if (
-      getMode(this.win).geoOverride &&
-      (isCanary(this.win) || getMode(this.win).localDev)
-    ) {
+    const geoOverride = getHashParams(this.win)['amp-geo'];
+    if (geoOverride && (isCanary(this.win) || getMode(this.win).localDev)) {
       // debug override case, only works in canary or localdev
       // match to \w characters only to prevent xss vector
       const overrideGeoMatch = GEO_HOTPATCH_STR_REGEX.exec(
-        getMode(this.win).geoOverride.toLowerCase()
+        geoOverride.toLowerCase()
       );
       if (overrideGeoMatch[1]) {
-        this.country_ = overrideGeoMatch[1].toLowerCase();
+        this.country_ = overrideGeoMatch[1];
```

Comment:
The input string to the regex is lowercase, so the matches are too.

Reply:
I don't think this is a valid change. Even though it still matches, there could be other pieces of code that assume `this.country_` was set to lowercase right?

Reply:
That doesn't matter; `overrideGeoMatch` stores the regex captures groups from running `/^(?:(\w{2})(?:\s(\w{2}-\w{2}))?)?\s*/` against `geoOverride.toLowerCase()`. That means these values are guaranteed to be substrings of an already-lowercased string. Calling `.toLowerCase()` does nothing

Reply:
Oh I see. I had missed the `geoOverride.toLowerCase()` bit. LGTM

Reply:
These lowercases should be added back. We don't control all servers that do hot-patching, and even we screwed this up [once upon a time](http://cl/311364776)